packages/design-systems/src/components/composite/Datagrid/useDataGrid.tsx

```diff
@@ -83,6 +88,27 @@ export const useDataGrid = <TData extends RowLike>({
         size: 54,
       });
   }
+  const state = manualPagination
+    ? {
+        columnVisibility,
+        columnPinning: columnPinningState,
+        rowSelection: enableRowSelection ? rowSelection : {},
+        exportOptions,
+        sorting: sorting,
+      }
+    : {
+        columnVisibility,
+        columnPinning: columnPinningState,
+        rowSelection: enableRowSelection ? rowSelection : {},
+        exportOptions,
+      };
+  const sortingConfigs = manualPagination
+    ? {
+        manualSorting: manualPagination,
+        onSortingChange: setSorting,
+      }
+    : {};
+
```

Comment:
Small style things, but I guess we can do this here. If-condition on `manualPagition` here should be better to be single.

```suggestion
  const { state, sortingConfigs } = manualPagination
    ? {
       state: {
          columnVisibility,
          columnPinning: columnPinningState,
          rowSelection: enableRowSelection ? rowSelection : {},
          exportOptions,
          sorting: sorting,
        },
        sortingConfigs: {
          manualSorting: manualPagination,
          onSortingChange: setSorting,     
        }
      }
    : {
        state: {
          columnVisibility,
          columnPinning: columnPinningState,
          rowSelection: enableRowSelection ? rowSelection : {},
          exportOptions,
        },
        sortingConfigs: {}
      }
```